README.md

```diff
@@ -1,31 +1,617 @@
+
+`Polymorphic` provides method `createPolymorphicInfo()` which returns `GenericPolymorphicInfo`. This method has optional parameter `name` which represents the polymorphic name of the type (default value is real name of the type). `GenericPolymorphicInfo` allows you to register subtypes with overloads of `register()` and `with()` (`with()` returns `self` to allow chaining).
+
+**Example**
+
+```Swift
+class A: Polymorphic {
+
+    class var polymorphicKey: String {
+        return "K"
+    }
+
+    class var polymorphicInfo: PolymorphicInfo {
+        return createPolymorphicInfo(name: "Base").with(subtypes: B.self, D.self)
+    }
+}
+
+class B: A {
+
+    override class var polymorphicInfo: PolymorphicInfo {
+        return createPolymorphicInfo().with(subtype: C.self)
+    }
+}
+
+class C: B {
+
+	override class var polymorphicKey: String {
+        return "C"
+    }
+}
+
+class D: C {
+
+    override class var polymorphicInfo: PolymorphicInfo {
+        return createPolymorphicInfo()
+    }
+}
+```
+
+Note: This example omits implementation of Map protocol.
+
+There are few things to notice.
+
+1. `C` overrides `polymorphicKey` that means: `A` and `B` have key "K" and `C` and `D` have key "C". So `SupportedType.dictionary(["C": .string("C")]` represents `C` but `SupportedType.dictionary(["K": .string("C")]` means nothing in this context.
+1. `A` has explicit name "Base". So `SupportedType.dictionary(["K": .string("Base")]` represents `A`.
+1. `C` does not override `polymorphicInfo`. This means that `C` cannot be used as the input type (exception will be raised) but `D` can be, even though it won't ever resolve to another type.
+1. `D` is registered in `A` not `B`. Because of that `B` does not know about `D`. So if `B` is the input type you can never get `D` as subtype.
+1. `A` knows about `C` because it is register in `B` which is registered in `A`.
+
+### Thread safety
+
+DataMapper is designed to be used on background thread (default implementation is thread safe). If you want to use it that way you need to make sure that implementations of all methods from Map protocols are thread safe as well (or that the objects you are using cannot be used at two threads simultaneously). Your custom implementations of protocols like `Serializer`, `Polymorph`, `Transformation` etc. must be thread safe too.
+
+## Versioning
+
+This library uses semantic versioning. Until version 1.0 API breaking changes may occur even in minor versions. We consider version 0.1 to be prerelease, which means that API should be stable but is not tested yet in real project. After that testing we make needed adjustments and bump the version to 1.0 (first release).
```

Comment:
Until the version 1.0
We consider the version 0.1
not tested yet in THE/A real project
After that testing, we make necessary adjustments and